### PR TITLE
Support Slice, StridedSlice

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@huningxin, PTAL, thanks! I've verified it with a local build that applied the CL of the Slice implementation.

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc [hunk 1]
```diff
@@ -2832,6 +3014,156 @@ class Subgraph {
...
+    const TfLiteTensor& input_tensor = tensors[input_tensor_id];
+    const TfLiteTensor& begin_tensor = tensors[begin_tensor_id];
+    const TfLiteTensor& end_tensor = tensors[end_tensor_id];
+    const TfLiteTensor& output_tensor = tensors[output_tensor_id];
+
+    TF_LITE_ENSURE_STATUS(CheckShapeTensorShape(logging_context, begin_tensor,
+                                                begin_tensor_id, node_index));
+    TF_LITE_ENSURE_STATUS(CheckTensorStaticAllocation(
+        logging_context, begin_tensor, begin_tensor_id, node_index));
+    // TODO(b/246969669): TFLite only supports int32 begin ends and strides,
```
Comment (huningxin) [inline tensorflow/lite/delegates/webnn/webnn_delegate.cc]: What does `b/246969669` mean?

**Review by huningxin — APPROVED** [review]
LGTM with an open.

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc (hunk 1 above)
Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc]: Umm, seems a TFLite internal bug jira link, I just referred from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc#L5404.

Comment (huningxin) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc]: Consider posting a pointer to xnnpack_delegate source?

Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc]: Sure. Done.

Comment (huningxin) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc]: The pointer looks good. Please remove `b/246969669` within the TODO.

Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc]: Done.:)

**Review by huningxin — APPROVED** [review]

**Pull request merged**